### PR TITLE
node addons

### PR DESCRIPTION
**Pull request by mapsam**
Working on grabbing information from node-cpp-skel into a `node-cpp.md` file here. Still much to add, but the groundwork is laid! 

Refs: #6 

cc @springmeyer @GretaCB @mikemorris

### CODE REVIEW

**Comment by mapsam** [comment]
Ready for a review here @GretaCB @springmeyer! There is a bit of overlap from the README so it might be worth cleaning that up and linking to this document.

File: node-cpp.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# Node.js C++ Addons
+
+The following document outlines Mapbox's approach to writing native abstractions for Node.js (anecdotally referred to as _addons_).
```
Comment (springmeyer) [inline node-cpp.md]: `native abstractions for Node.js` is a description of [Nan](https://github.com/nodejs/nan) not node addons, so I would change to `The following document outlines Mapbox's approach to writing C++ modules for Node.js (often referred to as _addons_)`
- A node C++ addon is a module, writting in C++ that extends nodejs with functionality
- Nan is a header that makes it easier to write a c++ addon.
- You can write a C++ addon without Nan.

Comment (mapsam) [reply node-cpp.md]: Ah, good catch!

**Comment by mapsam** [comment]
Merging this as-is for now, let's edit as we go!

**Pull request merged**